### PR TITLE
DR action readiness check modification

### PR DESCRIPTION
**Pull request by GowthamShanmugam**
BZ: https://bugzilla.redhat.com/show_bug.cgi?id=2183153

### CODE REVIEW

File: packages/mco/components/modals/app-failover-relocate/argo-application-set.tsx [hunk 1]
```diff
@@ -122,7 +122,7 @@ export const ArogoApplicationSetModal = (
             targetClusterAvailableTime:
               targetClusterCondition?.lastTransitionTime,
             isPrimaryClusterAvailable: !!primaryClusterCondition,
-            isPeerReady: isPeerReadyAndAvailable(drPlacementControl),
+            isDRActionReady: checkDRActionReadiness(drPlacementControl, action),
```
Comment (SanjalKatiyar) [inline packages/mco/components/modals/app-failover-relocate/argo-application-set.tsx:125]: need to add `action` to the dependency list...

File: packages/mco/components/modals/app-failover-relocate/failover-relocate-modal-body.tsx [hunk 2]
```diff
@@ -75,11 +75,11 @@ const validatePlacement = (
     return isFailoverAction
       ? ErrorMessageType.DR_IS_NOT_ENABLED_FAILOVER
       : ErrorMessageType.DR_IS_NOT_ENABLED_RELOCATE;
-  } else if (!placement?.isPeerReady) {
+  } else if (!placement?.isDRActionReady) {
     // Peer is not ready for DR action
```
Comment (SanjalKatiyar) [inline packages/mco/components/modals/app-failover-relocate/failover-relocate-modal-body.tsx:79]: // Either Peer is not ready for DR failover
// Or, Peer is not ready/available for DR relocate

File: packages/mco/components/modals/app-failover-relocate/subscriptions/peer-cluster-status.tsx [hunk 3]
```diff
@@ -132,15 +140,16 @@ export const PeerClusterStatus: React.FC<PeerClusterStatusProps> = ({
   React.useEffect(() => {
     if (!!state.selectedSubsGroups.length) {
       const peerCurrentStatus = getPeerStatusSummary(
-        state?.drPolicyControlState,
-        state?.selectedSubsGroups,
+        drPolicyControlState,
+        selectedSubsGroups,
+        actionType,
```
Comment (SanjalKatiyar) [inline packages/mco/components/modals/app-failover-relocate/subscriptions/peer-cluster-status.tsx:145]: need to add `state` or `state.selectedSubsGroups.length` to the dependency list...

**Comment by SanjalKatiyar** [comment]
> BZ: https://bugzilla.redhat.com/show_bug.cgi?id=2176179

we need to update the bug, 2176179 is for 4.12

File: packages/mco/components/modals/app-failover-relocate/argo-application-set.tsx (hunk 1 above)
Comment (GowthamShanmugam) [reply packages/mco/components/modals/app-failover-relocate/argo-application-set.tsx:125]: done

File: packages/mco/components/modals/app-failover-relocate/failover-relocate-modal-body.tsx (hunk 2 above)
Comment (GowthamShanmugam) [reply packages/mco/components/modals/app-failover-relocate/failover-relocate-modal-body.tsx:79]: ack

**Comment by SanjalKatiyar** [comment]
> /approve

not adding "lgtm" as BZ do not have all the acks, this is anyway a "blocker" BZ and can be taken in after dev freeze as well...

**Pull request merged**